settings.cmake

```diff
@@ -139,7 +139,9 @@ if(NOT Sel4testAllowSettingsOverride)
     endif()
 
     if(SMP)
-        set(KernelMaxNumNodes 4 CACHE STRING "" FORCE)
+        if("${KernelMaxNumNodes}" STREQUAL "")
```

Comment:
Are you able to achieve this by allowing `SMP=ON` to imply 4 nodes, and `SMP=8` imply 8 nodes and `SMP=OFF` 1 node?

Reply:
Given that CMake is embraces the idea of truthiness (e.g `SMP=1` is equivalent to `SMP=ON`) and to provide backwards compatible behaviour, I've added a new var (`NUM_NODES`) which can be used to override the default of 4. Let me know what you think.

Reply:
Yea that's fine.  But you will also need to add `NUM_NODES` to easy_settings.cmake also

Reply:
cool, done